src/renderers/materialized/materialized.text.control.tsx

```diff
@@ -1,17 +1,46 @@
 import { JSX } from '../JSX';
 import { withIncreasedRank } from '../../core/testers';
 import { Control, ControlProps, ControlState } from '../controls/Control';
+import { resolveSchema } from '../../path.util';
+import { ControlElement } from '../../models/uischema';
 import { mapStateToControlProps, registerStartupRenderer } from '../renderer.util';
 import { textControlTester } from '../controls/text.control';
 import { connect, Event } from '../../common/binding';
-
+declare let $;
 export class MaterializedTextControl extends Control<ControlProps, ControlState> {
 
+  componentDidMount() {
+    const { id, uischema, schema } = this.props;
+    const controlElement = uischema as ControlElement;
+    const maxLength = resolveSchema(schema, controlElement.scope.$ref).maxLength;
+    if (uischema.options && uischema.options.trim && maxLength !== undefined) {
+      const input = $(`[id="${id}"]`);
+      const fontSize = parseFloat(input.css('font-size'));
+
+      /* The widest letter of the latin alphabet is W and has a width of 15px when
+         displayed with a font-size of 14.5px.
+         For the calculation of the input's width, the maximum number of allowed characters
+         (maxLength) has to be multiplied by 15px, as an input text consisting of W only
+         can be assumed to be the widest input text possible.
+         Furthermore, the result of this has to be multiplied by the ratio with which the
+         font-size has increased (decreased) compared to the base font-size of 14.5px
+         that applies to the mentioned width of 15px for a W, in order to enlarge (shrink)
+         the input width according to the changed font-size.
+       */
+      input.css('width', `${(maxLength * 15) * (fontSize / 14.5)}px`);
```

Comment:
didn't we agree to move the constants to two constant fields and add a jsdoc to each constant?

Reply:
@eneufeld sorry, I forgot about that. Now it's changed